File: data-structures/java/data-structures/src/main/java/com/data_structures_2/list/doubly_linked_list/IDoublyLinkedListNode.java
package com.data_structures_2.list.doubly_linked_list;

public interface IDoublyLinkedListNode<T> {

    T getData();

    void setData(T data);

    IDoublyLinkedListNode<T> getNext();

    void setNext(IDoublyLinkedListNode<T> next);

    IDoublyLinkedListNode<T> getPrev();

    void setPrev(IDoublyLinkedListNode<T> prev);
}

File: data-structures/python/data-structures/src/list/linked_list/doubly_linked_list/IDoublyLinkedListNode.py
from abc import ABC


class IDoublyLinkedListNode(ABC):
    @property
    def data(self):
        return

    @data.setter
    def data(self, val):
        pass

    @data.deleter
    def data(self):
        pass

    @property
    def prev(self):
        return

    @prev.setter
    def prev(self, val):
        pass

    @prev.deleter
    def prev(self):
        pass

    @property
    def next(self):
        return

    @next.setter
    def next(self, val):
        pass

    @next.deleter
    def next(self):
        pass
